components/ai_chat/core/common/mojom/untrusted_frame.mojom

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module ai_chat.mojom;
+
+// Interfaces for communication between the untrusted content frame and both
+// the Browser and the parent trusted UI frame.
+
+// Trusted WebUI-side handler for messages from the untrusted child frame
+interface ParentUIFrame {
+  // <iframe> cannot be sized to the natural height of its content, so the child
+  // must let the parent know whenever the body height changes so that the parent
+  // can manually set the same height on the <iframe> to avoid scrolling.
+  ChildHeightChanged(uint32 height);
+
+  // This is sent to the UI rather than straight to the browser to allow
+  // for further rating feedback to be attached. The parent frame deals
+  // with it instead of the child frame, due to UI space as well as sensitive
+  // user information being captured.
+  RateMessage(string turn_uuid, bool is_liked);
+};
+
+// Browser-side handler for untrusted frame that handles rendering of
+// conversation entries.
+interface UntrustedUIHandler {
+  OpenSearchURL(string query);
+  OpenLearnMoreAboutBraveSearchWithLeo();
+  BindParentPage(pending_receiver<ParentUIFrame> parent_frame);
+};
```

Comment:
These are the API calls that the Untrusted frame can make to the C++ UI class and the Javascript parent frame